### PR TITLE
HPCC-20700 XRefNode not found

### PR DESCRIPTION
**Pull request by kunalaswani**
XREF component file tabs disable unless files are generated.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by kunalaswani** [comment]
@miguelvazq please review

**Comment by miguelvazq** [comment]
You will have to change the "includes" to something similar to indexOf since IE does not support "includes"

**Comment by kunalaswani** [comment]
@miguelvazq changed and tested to ensure logic works as expected.

**Comment by kunalaswani** [comment]
@AttilaVamos can you take a look at this smoketest?

File: esp/src/eclwatch/XrefDetailsWidget.js [hunk 1]
```diff
@@ -105,7 +105,21 @@ define([
...
+                    this.setDisabled(this.widget._Directories.id, false);
+                    this.setDisabled(this.widget._Errors.id, false);
+                } else {
+                    this.setDisabled(this.widget._FoundFiles.id, true);
+                    this.setDisabled(this.widget._OrphanFiles.id, true);
+                    this.setDisabled(this.widget._LostFiles.id, true);
+                    this.setDisabled(this.widget._Directories.id, true);
+                    this.setDisabled(this.widget._Errors.id, true);
+                };
+
```
Comment (miguelvazq) [inline esp/src/eclwatch/XrefDetailsWidget.js]: extra space

**Review by miguelvazq — APPROVED** [review]

**Comment by AttilaVamos** [comment]
@kunalaswani Sorry we had a meeting. I found the problem:
```
npm test
	res:
	> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-12233/HPCC-Platform/esp/src
	> run-s lint
	> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-12233/HPCC-Platform/esp/src
	> jshint --config ./.jshintrc ./eclwatch
	eclwatch/XrefDetailsWidget.js: line 121, col 18, Unnecessary semicolon.
	1 error
	npm ERR! code ELIFECYCLE
	npm ERR! errno 2
	npm ERR! eclwatch@1.0.0 lint: `jshint --config ./.jshintrc ./eclwatch`
	npm ERR! Exit status 2
	npm ERR! 
	npm ERR! Failed at the eclwatch@1.0.0 lint script.
	npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
	npm WARN Local package.json exists, but node_modules missing, did you mean to install?
	npm ERR! A complete log of this run can be found in:
	npm ERR!     /mnt/disk1/home/vamosax/.npm/_logs/2019-02-27T16_48_09_789Z-debug.log
	ERROR: "lint" exited with 2.
	npm ERR! Test failed.  See above for more details.
npm test end
```
Usually this kind of error is reported by Smoketest. I will check and fix it soon.

**Comment by AttilaVamos** [comment]
@kunalaswani please do not fix it yet. I would fix the Smoketest to report this kind of problem again.

**Comment by AttilaVamos** [comment]
@kunalaswani Thanks for your help. Now you can fix this problem.

**Comment by kunalaswani** [comment]
@AttilaVamos thanks! Changes made.

**Comment by AttilaVamos** [comment]
@kunalaswani Hmm, something is still not right with Smoketest. I will fix it and reschedule this PR.

**Comment by kunalaswani** [comment]
@AttilaVamos this looks safe to merge, i fixed my issue, can you confirm smoke test looks fine?

**Comment by AttilaVamos** [comment]
@kunalaswani it is fine.

**Pull request merged**